static/app/views/issueList/groupSearchViewTabs/issueViews.tsx

```diff
@@ -0,0 +1,548 @@
+
+      if (viewId?.startsWith('_') && currentView) {
+        const matchingView = newlyCreatedViews.find(
+          view =>
+            view.id &&
+            currentView.query === view.query &&
+            currentView.querySort === view.querySort
+        );
+        if (matchingView?.id) {
+          navigate(
+            normalizeUrl({
+              ...location,
+              query: {
+                ...queryParamsWithPageFilters,
+                viewId: matchingView.id,
+              },
+            }),
+            {replace: true}
+          );
+        }
+      }
+    }
+    return;
+  };
+
+  const {mutate: updateViews} = useUpdateGroupSearchViews({
+    onSuccess: replaceWithPersistantViewIds,
+  });
+
+  const debounceUpdateViews = useMemo(
+    () =>
+      debounce((newTabs: IssueView[]) => {
+        if (newTabs) {
+          updateViews({
+            orgSlug: organization.slug,
+            groupSearchViews: newTabs
+              .filter(tab => tab.isCommitted)
+              .map(tab => ({
+                // Do not send over an ID if it's a temporary or default tab so that
+                // the backend will save these and generate permanent Ids for them
+                ...(tab.id[0] !== '_' && !tab.id.startsWith('default')
+                  ? {id: tab.id}
+                  : {}),
+                name: tab.label,
+                query: tab.query,
+                querySort: tab.querySort,
+              })),
+          });
+        }
+      }, 500),
+    [organization.slug, updateViews]
+  );
+
+  const reducer: Reducer<IssueViewsState, IssueViewsActions> = useCallback(
+    (state, action): IssueViewsState => {
+      if (!tabListState) {
+        return state;
+      }
+      switch (action.type) {
+        case 'REORDER_TABS':
+          return reorderTabs(state, action);
+        case 'SAVE_CHANGES':
+          return saveChanges(state, tabListState);
+        case 'DISCARD_CHANGES':
+          return discardChanges(state, tabListState);
+        case 'RENAME_TAB':
+          return renameView(state, action, tabListState);
+        case 'DUPLICATE_VIEW':
+          return duplicateView(state, action, tabListState);
+        case 'DELETE_VIEW':
+          return deleteView(state, tabListState);
+        case 'CREATE_NEW_VIEW':
+          return createNewView(state, action);
+        case 'SET_TEMP_VIEW':
+          return setTempView(state, action);
+        case 'DISCARD_TEMP_VIEW':
+          return discardTempView(state, tabListState);
+        case 'SAVE_TEMP_VIEW':
+          return saveTempView(state, tabListState);
+        case 'UPDATE_UNSAVED_CHANGES':
+          return updateUnsavedChanges(state, action, tabListState);
+        case 'UPDATE_VIEW_IDS':
+          return updateViewIds(state, action);
+        case 'SET_VIEWS':
+          return setViews(state, action);
+        case 'SYNC_VIEWS_TO_BACKEND':
+          debounceUpdateViews(state.views);
```

Comment:
Does this need to be in the reducer?

Reply:
Oop no i meant to delete that, good catch!